### ISSUE TITLE
Doc page "AJAX and APIs" shows invalid code snippets which will throw a memory leak error

### ISSUE THREAD

**Issue by SpacePotatoBear** [reactjs/react.dev#1068]
The page https://reactjs.org/docs/faq-ajax.html shows a fetch followed by a this.setState in the '.then'

This will throw "Warning: Can't call setState (or forceUpdate) on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method."

This is because the fetch call may complete after the component is unmounted. 

The fetch api now supports cancellation (https://developers.google.com/web/updates/2017/09/abortable-fetch) via a controller but reactjs doesn't yet support this.

Thus the docs should be updated to include the "anti paterns" of this.isMounted to guard the set state, use a 3rd party fetch API like Axios which supports cancellation or have this issue pushed to the reactjs team to implement abort able fetch && have docs updated to reflect that.

**Comment by alexkrolick** [comment]
> The fetch api now supports cancellation via a controller but reactjs doesn't yet support this

React doesn't provide `fetch`, it's provided by the browser. Doesn't look like the feature is supported in very many browsers though.

@gaearon there was some text about this but the error message was wrong

```diff
diff --git a/38a532fe b/bdd3ad7a
index 38a532fe..bdd3ad7a 100644
--- a/38a532fe
+++ b/bdd3ad7a
@@ -82,7 +82,3 @@ class MyComponent extends React.Component {
   }
 }

-
-### Cancellation
-
- Note that if the component unmounts before an AJAX call is complete,
- you may see a warning like `cannot read property 'setState' of undefined`.
- If this is an issue you maywant to keep track of inflight AJAX requests and
- cancel them in the `componentWillUnmount` lifecycle method.
```

**Comment by SpacePotatoBear** [comment]
I mean react doesn't support it because trying to create an abort controller as defined in the google page or https://developer.mozilla.org/en-US/docs/Web/API/AbortController/AbortController gives this error 

``  Line 12:  'AbortController' is not defined  no-undef

Search for the keywords to learn more about each error.``

**Comment by adeelibr** [comment]
Can i work on this, and make A PR in the faqs section. I have worked with Abort Controllers in react.

**Comment by adeelibr** [comment]
As a reference I wrote an article about Abort Controllers in react https://medium.freecodecamp.org/how-to-work-with-react-the-right-way-to-avoid-some-common-pitfalls-fc9eb5e34d9e

**Comment by alexkrolick** [comment]
Yes, go ahead 😄. If you change your mind, check back here and we'll remove the in-progress label.

**Comment by adeelibr** [comment]
I am working on it.

**Comment by BartoszKlonowski** [comment]
From what I can see we no longer have the AJAX page in the latest version of docs, which makes this issue no longer valid.
Let me close this as not planned, but please reach out in case of doubts/issues.
